polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -1345,6 +1387,11 @@ void LoopInternalization::transform(
   createWorkGroupBarrier(builder);
   builder.setInsertionPointAfter(loop);
   createWorkGroupBarrier(builder);
+
+  // When work group size is unknown at compile time, unroll the tiled loop to
```

Comment:
One note here (also previously discussed with @whitneywhtsang). We could teach mlir how to handle loop directives. The LLVM dialect has support for loop unrolling already, however the affine and scf dialect do not. Also the control -flow dialect would need support as well. 

Once support for handling loop directive is present in these mlir dialects we might want to remove this code and just attach an attribute to indicate the loop should be unrolled by a factor to 4.